back/engines/free/user_confirmation/app/interactors/user_confirmation/confirm_user.rb

```diff
@@ -0,0 +1,32 @@
+module UserConfirmation
+  class ConfirmUser < ApplicationInteractor
+    delegate :user, :code, to: :context
+
+    def call
+      validate_user
+      validate_code
+      confirm_user
+    end
+
+    def validate_user
+      fail_with_error! :user, :blank if user.blank?
```

Comment:
this adds `ActiveMode::Errors` to the interactor `result`, meaning you can do `result.errors.details` in the controller.